keanu-project/src/main/java/io/improbable/keanu/tensor/dbl/Nd4jDoubleTensor.java

```diff
@@ -171,6 +171,24 @@ public double standardDeviation() {
             .sumNumber().doubleValue() / (tensor.length() - 1));
     }
 
+    @Override
+    public boolean equalsWithEps(DoubleTensor o, double epsilon) {
+        if (this == o) return true;
+
+        if (o instanceof Nd4jDoubleTensor) {
+            return tensor.equalsWithEps(((Nd4jDoubleTensor) o).tensor,epsilon);
+        } else {
+            if (this.hasSameShapeAs(o)) {
+                DoubleTensor difference = o.minus(this);
+                for (double v : difference.asFlatList()) {
```

Comment:
`difference.getGreaterThanMask(DoubleTensor.scalar(epsilon)).sum() >= 1` or `difference.lessThan(epsilon).allTrue()`. Not sure which is more performant but either would be preferable here.